nsqd/stats.go

```diff
@@ -48,6 +48,11 @@ type ChannelStats struct {
 }
 
 func NewChannelStats(c *Channel, clients []ClientStats) ChannelStats {
+	c.inFlightMutex.Lock()
+	defer c.inFlightMutex.Unlock()
+	c.deferredMutex.Lock()
+	defer c.deferredMutex.Unlock()
```

Comment:
Let's tighten up the critical sections between these locks and move the `len(...)` reads from out of the struct instantiation to temp variables here?